src/wasm-traversal.h

```diff
@@ -256,18 +256,18 @@ struct Walker : public VisitorType {
         self->walkTag(curr.get());
       }
     }
-    for (auto& curr : module->elementSegments) {
-      self->walkElementSegment(curr.get());
-    }
     for (auto& curr : module->tables) {
       self->walkTable(curr.get());
     }
-    for (auto& curr : module->dataSegments) {
-      self->walkDataSegment(curr.get());
+    for (auto& curr : module->elementSegments) {
+      self->walkElementSegment(curr.get());
     }
     for (auto& curr : module->memories) {
       self->walkMemory(curr.get());
     }
+    for (auto& curr : module->dataSegments) {
+      self->walkDataSegment(curr.get());
+    }
```

Comment:
I actually think we should keep this changes, as I said before. They are postorder-like.

But that is separate from the idea of not depending on the order in the pass, which I am not opposed to.